op-supervisor/supervisor/backend/cross/safe_frontier_test.go

```diff
@@ -11,6 +11,55 @@ import (
 	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
 )
 
+func TestSufficientDataChecks(t *testing.T) {
```

Comment:
One more test-case I think we need: make the first cross-safe lookup return a cross-safe with sufficient source, but the second cross-safe lookup return an error, with a depset of 2. And sanity check the chainID arg is different. To ensure it really is iterating over the hazards.